### DISCUSSION TITLE
Accessing Flask context processors in components

### DISCUSSION THREAD

**Discussion by jumboduck** [jpsca/jinjax#83] [Preguntas/Respuestas] [answered]
I've been playing around with Jinjax to understand if it would work with one of my Flask projects.
A big issue I'm running into is the ability to access context processors from the components. An example of this would be for example using the `current_user` object from `Flask-Login`.

Is there something I'm overlooking?

**Comment by stfnx** [comment]
Hey, I just ran into the same problem.
Did you find a solution for this?

The Flask instance I simply added to Jinja globals like this:
```
app = Flask(__name__)
app.jinja_env.globals.update({
    "app": app
})
```

And then stuff like `{% if app.debug %}` does work in JinjaX components.

But I think the same wouldn't work for `current_user` from Flask-Login.

**Comment by jpsca** [comment, marked as answer]
You can tell JinjaX to use the `app.jinja_env` instead of creating a new one, like this:

```python
app = Flask(__name__)
catalog = jinjax.Catalog(jinja_env=app.jinja_env)
```

so everything Flask or any extension adds to Jinja environment, like the `current_user`, will be available inside JinjaX templates
